Common/Data/Helper/Sync/SyncEngine.swift

```diff
@@ -105,11 +105,8 @@ public class SyncEngine {
 
     private func buildCreateRequest(forQuery query: ResourceURLRepresentable,
                                     forResource resource: Pushable) -> Result<URLRequest, XikoloError> {
-        switch resource.resourceData() {
-        case let .success(resourceData):
+        return resource.resourceData().flatMap { resourceData in
```

Comment:
How does this part work?

Reply:
What du you mean? It's basically the same construct, only shorter.